geomstats/geometry/general_linear.py

```diff
@@ -1,192 +1,125 @@
 """Module exposing the GeneralLinear group class."""
 
 import geomstats.backend as gs
-from geomstats.geometry.lie_group import LieGroup
 from geomstats.geometry.matrices import Matrices
 
 
-class GeneralLinear(LieGroup, Matrices):
-    """Class for the General Linear Group, i.e. the matrix group GL(n).
-
-    Note: The default representation for elements of GL(n)
-    are matrices. For now, SO(n) and SE(n) elements are represented
-    by a vector by default.
-    """
+class GeneralLinear(Matrices):
+    """Class for the general linear group GL(n)."""
 
     def __init__(self, n):
-        assert isinstance(n, int) and n > 0
-        LieGroup.__init__(self, dimension=n*n)
-        Matrices.__init__(self, m=n, n=n)
-
-    def get_identity(self, point_type=None):
-        """Compute the identity of the General Linear Group.
-
-        Parameters
-        ----------
-        point_type : TODO
-
-        Returns
-        -------
-        identity : TODO
+        self.n = n
+
+    @staticmethod
+    def belongs(point):
+        """Test if a matrix is invertible."""
+        det = gs.linalg.det(point)
+        return gs.where(det != 0., True, False)
```

Comment:
I wonder if we should use something like ` not gs.isclose(det, 0.)`, in order to be flexible with respect to numerical imprecisions (if the determinant is 1e-5, should we consider it is 0.?)

Reply:
I guess we could take the benefit of avoiding where!
Matrices should be invertible as soon `det != 0.` though.

Reply:
We don't have a vectorized working `not`  :( 
We have to add logical operations to the backend...